invenio_indexer/api.py

```diff
@@ -271,24 +277,33 @@ def _index_action(self, payload):
         """
         record = Record.get_record(payload['id'])
         index, doc_type = self.record_to_index(record)
+        arguments = {}
 
-        return {
+        body = self._prepare_record(record, index, doc_type, arguments)
+
+        action = {
             '_op_type': 'index',
             '_index': index,
             '_type': doc_type,
             '_id': str(record.id),
             '_version': record.revision_id,
             '_version_type': self._version_type,
-            '_source': self._prepare_record(record, index, doc_type),
+            '_source': body
         }
 
+        action.update(arguments)
+
+        return action
+
     @staticmethod
-    def _prepare_record(record, index, doc_type):
+    def _prepare_record(record, index, doc_type, arguments, **kwargs):
```

Comment:
To keep backward compatibility (even though this is a "private" method), we could instead do `arguments=None`, and then inside the method `arguments = {} if arguments is None else arguments`.